scm/github.go

```diff
@@ -661,6 +658,85 @@ func (s *GithubSCM) UpdateIssueComment(ctx context.Context, opt *IssueCommentOpt
 	return nil
 }
 
+// CreateCourse creates repositories and teams for a new course.
+func (s *GithubSCM) CreateCourse(ctx context.Context, opt *NewCourseOptions) ([]*Repository, error) {
+	// Get and check the organization's suitability for the course
+	org, err := s.GetOrganization(ctx, &GetOrgOptions{ID: opt.OrganizationID, NewCourse: true})
+	if err != nil {
+		return nil, err
+	}
+
+	// Set restrictions to prevent students from creating new repositories and prevent access
+	// to organization repositories. This will not affect organization owners (teachers).
+	defaultPermissions := OrgNone
+	createRepoPermissions := false
+	if _, _, err = s.client.Organizations.Edit(ctx, org.Name, &github.Organization{
```

Comment:
```go
	// DefaultRepoPermission can be one of: "read", "write", "admin", or "none". (Default: "read").
	// It is only used in OrganizationsService.Edit.
	DefaultRepoPermission *string `json:"default_repository_permission,omitempty"`
	// MembersCanCreateRepos default value is true and is only used in Organizations.Edit.
	MembersCanCreateRepos *bool `json:"members_can_create_repositories,omitempty"`
```
Seems like `DefaultRepoPermission` is not supported by `Organizations.Edit` only `OrganizationsService.Edit`. Thus, I'm not sure if this argument has any effect?? Can you check and add a comment if we still need to use the `DefaultRepoPermission` or alternatively, we maybe need to also invoke `OrganizationsService.Edit`??

Reply:
Never mind. This seems to be a type or weirdness on the part of the library. `OrganizationsService` is the type, while `Organizations` is a field of that type. So it is probably the same thing.